### PR TITLE
Add new API functions

### PR DESCRIPTION
**Pull request by ggrbill**
They are:
- get_number_of_fields
- get_number_of_phases
- get_number_of_layers
- get_number_of_phase_pairs

### CODE REVIEW

File: src/alfasim_sdk/alfasim_sdk_api/api.h [hunk 1]
```diff
@@ -273,6 +273,71 @@ DLL_EXPORT int get_plugin_variable(void* ctx, void** out, const char* variable_n
 */
 DLL_EXPORT int get_field_id(void* ctx, int* out, const char* name);
 
+/*!
+    Gets the phase ID of the given name. Althought this depends on the hydrodynamic model
```
Comment (arthursoprana) [inline src/alfasim_sdk/alfasim_sdk_api/api.h:277]:
```suggestion
    Gets the phase ID of the given name. Although this depends on the hydrodynamic model
```

File: src/alfasim_sdk/alfasim_sdk_api/api.h [hunk 2]
```diff
@@ -273,6 +273,71 @@ DLL_EXPORT int get_plugin_variable(void* ctx, void** out, const char* variable_n
...
+
+    @param[in] ctx ALFAsim's plugins context.
+    @param[out] out Phase ID.
+    @param[in] name Name of the phase to retrieve the ID.
+    @return An #error_code value.
+*/
+DLL_EXPORT int get_phase_id(void* ctx, int* out, const char* name);
+
+/*!
+    Gets the layer ID of the given name. Althought this depends on the hydrodynamic model
```
Comment (arthursoprana) [inline src/alfasim_sdk/alfasim_sdk_api/api.h:289]:
```suggestion
    Gets the layer ID of the given name. Although this depends on the hydrodynamic model
```

**Review by arthursoprana — APPROVED** [review]

**Review by prusse-martin — APPROVED** [review]

File: src/alfasim_sdk/alfasim_sdk_api/api.h (hunk 1 above)
Comment (ggrbill) [reply src/alfasim_sdk/alfasim_sdk_api/api.h]: Done

File: src/alfasim_sdk/alfasim_sdk_api/api.h (hunk 2 above)
Comment (ggrbill) [reply src/alfasim_sdk/alfasim_sdk_api/api.h]: Done

**Pull request merged**